guava/src/com/google/common/escape/CharEscaperBuilder.java

```diff
@@ -108,9 +113,14 @@ public CharEscaperBuilder addEscapes(char[] cs, String r) {
    *
    * @return a "sparse" array that holds the replacement mappings.
    */
+  @SuppressWarnings(value = {"lowerbound:array.access.unsafe.low",//Character types are non negative: https://github.com/kelloggm/checker-framework/issues/192
+          "upperbound:enhancedfor.type.incompatible"/* because `result` is a local variable declared inside of toArray()
+          Since `max` is maximum index is the value of the highest character that has been seen,
+          Key values in map can be indexed by the `result` array.
```

Comment:
This explanation is confusing. The clauses starting with "because" and "since" don't seem to correspond to the conclusion you draw, but maybe I just don't understand. Either way, this explanation needs to be improved so that it is more clear to the reader what's actually happening here.